### PR TITLE
HBASE-23170 Admin#getRegionServers use ClusterMetrics.Option.SERVERS_NAME

### PR DESCRIPTION
**Pull request by mymeiyi**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   2m  0s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| :blue_heart: |  mvndep  |   1m  6s |  Maven dependency ordering for branch  |
| :green_heart: |  mvninstall  |   5m 20s |  master passed  |
| :green_heart: |  compile  |   1m 23s |  master passed  |
| :green_heart: |  checkstyle  |   1m 52s |  master passed  |
| :green_heart: |  shadedjars  |   4m 34s |  branch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  javadoc  |   1m  1s |  master passed  |
| :blue_heart: |  spotbugs  |   4m  5s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| :green_heart: |  findbugs  |   5m  9s |  master passed  |
||| _ Patch Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 16s |  Maven dependency ordering for patch  |
| :green_heart: |  mvninstall  |   4m 56s |  the patch passed  |
| :green_heart: |  compile  |   1m 23s |  the patch passed  |
| :green_heart: |  javac  |   1m 23s |  the patch passed  |
| :green_heart: |  checkstyle  |   1m 49s |  the patch passed  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| :green_heart: |  shadedjars  |   4m 33s |  patch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  hadoopcheck  |  15m 32s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| :green_heart: |  javadoc  |   0m 59s |  the patch passed  |
| :green_heart: |  findbugs  |   5m 19s |  the patch passed  |
||| _ Other Tests _ |
| :green_heart: |  unit  |   1m 54s |  hbase-client in the patch passed.  |
| :green_heart: |  unit  | 165m 55s |  hbase-server in the patch passed.  |
| :green_heart: |  asflicense  |   1m  7s |  The patch does not generate ASF License warnings.  |
|  |   | 232m 55s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.3 Server=19.03.3 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-721/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/721 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 8acbf46eaed0 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-721/out/precommit/personality/provided.sh |
| git revision | master / 9e628b715d |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-721/1/testReport/ |
| Max. process+thread count | 4607 (vs. ulimit of 10000) |
| modules | C: hbase-client hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-721/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   0m 35s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 6 new or modified test files.  |
||| _ master Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 37s |  Maven dependency ordering for branch  |
| :green_heart: |  mvninstall  |   5m 48s |  master passed  |
| :green_heart: |  compile  |   1m 52s |  master passed  |
| :green_heart: |  checkstyle  |   2m 20s |  master passed  |
| :green_heart: |  shadedjars  |   4m 39s |  branch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  javadoc  |   1m 17s |  master passed  |
| :blue_heart: |  spotbugs  |   4m 11s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| :green_heart: |  findbugs  |   5m 21s |  master passed  |
||| _ Patch Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 16s |  Maven dependency ordering for patch  |
| :green_heart: |  mvninstall  |   4m 56s |  the patch passed  |
| :green_heart: |  compile  |   1m 53s |  the patch passed  |
| :green_heart: |  javac  |   1m 53s |  the patch passed  |
| :broken_heart: |  checkstyle  |   1m 25s |  hbase-server: The patch generated 7 new + 270 unchanged - 0 fixed = 277 total (was 270)  |
| :broken_heart: |  checkstyle  |   0m 18s |  hbase-it: The patch generated 6 new + 46 unchanged - 0 fixed = 52 total (was 46)  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| :green_heart: |  shadedjars  |   4m 36s |  patch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  hadoopcheck  |  15m 34s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| :green_heart: |  javadoc  |   1m 13s |  the patch passed  |
| :green_heart: |  findbugs  |   5m 18s |  the patch passed  |
||| _ Other Tests _ |
| :green_heart: |  unit  |   1m 53s |  hbase-client in the patch passed.  |
| :green_heart: |  unit  | 164m 26s |  hbase-server in the patch passed.  |
| :green_heart: |  unit  |   1m 10s |  hbase-it in the patch passed.  |
| :green_heart: |  asflicense  |   1m 38s |  The patch does not generate ASF License warnings.  |
|  |   | 234m 23s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.3 Server=19.03.3 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-721/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/721 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux f29654b2121c 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-721/out/precommit/personality/provided.sh |
| git revision | master / 6aec958d66 |
| Default Java | 1.8.0_181 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-721/2/artifact/out/diff-checkstyle-hbase-server.txt |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-721/2/artifact/out/diff-checkstyle-hbase-it.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-721/2/testReport/ |
| Max. process+thread count | 4585 (vs. ulimit of 10000) |
| modules | C: hbase-client hbase-server hbase-it U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-721/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/test/java/org/apache/hadoop/hbase/TestClientClusterMetrics.java [hunk 1]
```diff
@@ -104,6 +104,7 @@ public void testDefaults() throws Exception {
         defaults.getLiveServerMetrics().size());
     Assert.assertEquals(origin.getMasterInfoPort(), defaults.getMasterInfoPort());
     Assert.assertEquals(origin.getServersName().size(), defaults.getServersName().size());
+    Assert.assertEquals(ADMIN.getRegionServers().size(), defaults.getServersName().size());
```
Comment (infraio) [inline hbase-server/src/test/java/org/apache/hadoop/hbase/TestClientClusterMetrics.java]: Why add this?

File: hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestAsyncClusterAdminApi.java [hunk 2]
```diff
@@ -265,6 +265,7 @@ public void testGetRegionLoads() throws Exception {
 
     // Check RegionLoad matches the regionLoad from ClusterStatus
     ClusterMetrics clusterStatus = admin.getClusterMetrics(EnumSet.of(Option.LIVE_SERVERS)).get();
+    assertEquals(servers.size(), clusterStatus.getLiveServerMetrics().size());
```
Comment (infraio) [inline hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestAsyncClusterAdminApi.java]: Why add this assert?

File: hbase-server/src/test/java/org/apache/hadoop/hbase/TestClientClusterMetrics.java (hunk 1 above)
Comment (mymeiyi) [reply hbase-server/src/test/java/org/apache/hadoop/hbase/TestClientClusterMetrics.java]: Used for testing the Admin#getRegionServers method

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   0m 35s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 6 new or modified test files.  |
||| _ master Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 36s |  Maven dependency ordering for branch  |
| :green_heart: |  mvninstall  |   5m 46s |  master passed  |
| :green_heart: |  compile  |   1m 52s |  master passed  |
| :green_heart: |  checkstyle  |   2m 15s |  master passed  |
| :green_heart: |  shadedjars  |   4m 32s |  branch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  javadoc  |   1m 14s |  master passed  |
| :blue_heart: |  spotbugs  |   4m  5s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| :green_heart: |  findbugs  |   5m  8s |  master passed  |
||| _ Patch Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| :green_heart: |  mvninstall  |   4m 56s |  the patch passed  |
| :green_heart: |  compile  |   1m 52s |  the patch passed  |
| :green_heart: |  javac  |   1m 52s |  the patch passed  |
| :green_heart: |  checkstyle  |   2m 16s |  the patch passed  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| :green_heart: |  shadedjars  |   4m 36s |  patch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  hadoopcheck  |  15m 36s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| :green_heart: |  javadoc  |   1m 14s |  the patch passed  |
| :green_heart: |  findbugs  |   5m 19s |  the patch passed  |
||| _ Other Tests _ |
| :green_heart: |  unit  |   1m 51s |  hbase-client in the patch passed.  |
| :green_heart: |  unit  | 163m 10s |  hbase-server in the patch passed.  |
| :green_heart: |  unit  |   1m 10s |  hbase-it in the patch passed.  |
| :green_heart: |  asflicense  |   1m 38s |  The patch does not generate ASF License warnings.  |
|  |   | 232m 41s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.3 Server=19.03.3 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-721/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/721 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux b6169f76da09 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-721/out/precommit/personality/provided.sh |
| git revision | master / 7924ba39e7 |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-721/3/testReport/ |
| Max. process+thread count | 4976 (vs. ulimit of 10000) |
| modules | C: hbase-client hbase-server hbase-it U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-721/3/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncAdmin.java [hunk 3]
```diff
@@ -1048,8 +1048,7 @@
    * @return current live region servers list wrapped by {@link CompletableFuture}
    */
   default CompletableFuture<Collection<ServerName>> getRegionServers() {
-    return getClusterMetrics(EnumSet.of(Option.LIVE_SERVERS))
-      .thenApply(cm -> cm.getLiveServerMetrics().keySet());
+    return getClusterMetrics(EnumSet.of(Option.SERVERS_NAME)).thenApply(cm -> cm.getServersName());
```
Comment (Apache9) [inline hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncAdmin.java]: ClusterMetrics::getServersName?

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/RegionPlacementMaintainer.java [hunk 4]
```diff
@@ -204,9 +202,7 @@ private void genAssignmentPlan(TableName tableName,
 
     // Get the all the region servers
     List<ServerName> servers = new ArrayList<>();
-    servers.addAll(
-      FutureUtils.get(getConnection().getAdmin().getClusterMetrics(EnumSet.of(Option.LIVE_SERVERS)))
-        .getLiveServerMetrics().keySet());
+    servers.addAll(FutureUtils.get(getConnection().getAdmin().getRegionServers()));
```
Comment (Apache9) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/master/RegionPlacementMaintainer.java]: Not introduced by you, but I wonder whether we need to close the Admin here?

Comment (mymeiyi) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/master/RegionPlacementMaintainer.java]: It's a async admin and does not need to close.

File: hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncAdmin.java (hunk 3 above)
Comment (mymeiyi) [reply hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncAdmin.java]: done

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   0m 32s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 6 new or modified test files.  |
||| _ master Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 36s |  Maven dependency ordering for branch  |
| :green_heart: |  mvninstall  |   5m 16s |  master passed  |
| :green_heart: |  compile  |   1m 52s |  master passed  |
| :green_heart: |  checkstyle  |   2m 16s |  master passed  |
| :green_heart: |  shadedjars  |   4m 34s |  branch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  javadoc  |   1m 15s |  master passed  |
| :blue_heart: |  spotbugs  |   4m  3s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| :green_heart: |  findbugs  |   5m  8s |  master passed  |
||| _ Patch Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| :green_heart: |  mvninstall  |   5m  1s |  the patch passed  |
| :green_heart: |  compile  |   1m 57s |  the patch passed  |
| :green_heart: |  javac  |   1m 57s |  the patch passed  |
| :green_heart: |  checkstyle  |   2m 14s |  the patch passed  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| :green_heart: |  shadedjars  |   4m 44s |  patch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  hadoopcheck  |  15m 47s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| :green_heart: |  javadoc  |   1m 14s |  the patch passed  |
| :green_heart: |  findbugs  |   5m 20s |  the patch passed  |
||| _ Other Tests _ |
| :green_heart: |  unit  |   1m 53s |  hbase-client in the patch passed.  |
| :green_heart: |  unit  | 160m 50s |  hbase-server in the patch passed.  |
| :green_heart: |  unit  |   1m 12s |  hbase-it in the patch passed.  |
| :green_heart: |  asflicense  |   1m 23s |  The patch does not generate ASF License warnings.  |
|  |   | 230m 12s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.3 Server=19.03.3 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-721/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/721 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux f35c4b971bc6 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-721/out/precommit/personality/provided.sh |
| git revision | master / 0043dfebc5 |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-721/4/testReport/ |
| Max. process+thread count | 4709 (vs. ulimit of 10000) |
| modules | C: hbase-client hbase-server hbase-it U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-721/4/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**